datafusion/sqllogictest/test_files/union.slt

```diff
@@ -406,15 +406,16 @@ logical_plan
 physical_plan
 01)GlobalLimitExec: skip=0, fetch=5
 02)--SortPreservingMergeExec: [c9@1 DESC], fetch=5
-03)----UnionExec
-04)------SortExec: expr=[c9@1 DESC], preserve_partitioning=[true]
-05)--------ProjectionExec: expr=[c1@0 as c1, CAST(c9@1 AS Int64) as c9]
-06)----------RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
-07)------------CsvExec: file_groups={1 group: [[WORKSPACE_ROOT/testing/data/csv/aggregate_test_100.csv]]}, projection=[c1, c9], has_header=true
-08)------SortExec: expr=[c9@1 DESC], preserve_partitioning=[true]
-09)--------ProjectionExec: expr=[c1@0 as c1, CAST(c3@1 AS Int64) as c9]
-10)----------RepartitionExec: partitioning=RoundRobinBatch(4), input_partitions=1
-11)------------CsvExec: file_groups={1 group: [[WORKSPACE_ROOT/testing/data/csv/aggregate_test_100.csv]]}, projection=[c1, c3], has_header=true
+03)----LocalLimitExec: fetch=5
+04)------UnionExec
+05)--------SortExec: expr=[c9@1 DESC], preserve_partitioning=[true]
```

Comment:
I think SortExec can be limited -- which would likely be better to use too

https://docs.rs/datafusion-physical-plan/40.0.0/src/datafusion_physical_plan/sorts/sort.rs.html#683-684

I think you could also push the limit all the way down to the CSVExec in this plan too

Reply:
I actually would have expected the logical optimizer to have already pushed the limit into the inputs 🤔 


https://github.com/apache/datafusion/blob/fc8e7b90356b94af5f591240b8165bc4c8275a51/datafusion/optimizer/src/push_down_limit.rs#L103